src/installer.js

```diff
@@ -184,31 +165,31 @@ module.exports = (data, callback) => {
   data.rename = data.rename || defaultRename
   data.logger = data.logger || defaultLogger
 
-  let options
-
-  const promise = getDefaults(data)
-    .then(defaults => getOptions(data, defaults))
-    .then(generatedOptions => {
-      options = generatedOptions
-      const adjustedVersion = util.replaceInvalidVersionCharacters(options.version)
-      if (adjustedVersion !== options.version) {
-        data.logger('Warning: replacing disallowed characters in version to comply with SPEC format.' +
-          `Changing ${options.version} to ${adjustedVersion}`)
-        options.version = adjustedVersion
-      }
-      return data.logger(`Creating package with options\n${JSON.stringify(options, null, 2)}`)
-    }).then(() => common.createDir(options))
-    .then(dir => createMacros(options, dir))
-    .then(dir => createContents(options, dir))
-    .then(dir => createPackage(options, dir))
-    .then(dir => movePackage(options, dir))
+  if (callback) {
+    console.warn('The node-style callback is deprecated. In a future major version, it will be' +
+                 'removed in favor of a Promise-based async style.')
+  }
+
+  const installer = new RedhatInstaller(data)
+
+  const promise = installer.generateDefaults()
+    .then(() => installer.generateOptions())
+    .then(() => installer.generateScripts())
+    .then(() => data.logger(`Creating package with options\n${JSON.stringify(installer.options, null, 2)}`))
+    .then(() => installer.createStagingDir())
+    .then(() => installer.createMacros())
+    .then(() => installer.createContents())
+    .then(() => installer.createPackage())
+    .then(() => installer.movePackage())
     .then(() => {
-      data.logger(`Successfully created package at ${options.dest}`)
-      return options
+      data.logger(`Successfully created package at ${installer.options.dest}`)
+      return installer.options
     }).catch(err => {
       data.logger(common.errorMessage('creating package', err))
       throw err
     })
 
   return nodeify(promise, callback)
```

Comment:
We need to remove callback support. It's too bad that only now I realize about this, which could have been part of a previous release

Reply:
I think it's also in `-windows`.

Reply:
I also have `nodeify` in `electron-installer-snap`. I was planning on getting rid of that dependency the next time there's a major version bump, for `-snap` hopefully that's when Node 6 is EOL'd.